src/extensions/windows/Microsoft.DotNet.UpgradeAssistant.Extensions.Windows/WinformsDefaultFontUpdater.cs

```diff
@@ -32,9 +38,79 @@ public WinformsDefaultFontUpdater(ILogger<WinformsDefaultFontUpdater> logger)
             _logger = logger ?? throw new ArgumentNullException(nameof(logger));
         }
 
-        public Task<IUpdaterResult> ApplyAsync(IUpgradeContext context, ImmutableArray<IProject> inputs, CancellationToken token)
+        private string ProcessAppConfigFile(IProject project)
+        {
+            var hdpiValue = "SystemAware";
+            var appConfigFilePath = project.FindFiles("App.config").FirstOrDefault();
+            ConfigFile appConfigFile;
+            if (File.Exists(appConfigFilePath))
+            {
+                appConfigFile = new ConfigFile(appConfigFilePath);
+                var highDPISetting = appConfigFile.Contents.XPathSelectElement(HighDPIConfiguration);
+                if (highDPISetting is not null)
+                {
+                    var hdpi = highDPISetting.Elements("add").Where(e => e.Attribute("key").Value == "DpiAwareness").FirstOrDefault();
+                    if (hdpi is not null)
+                    {
+                        hdpiValue = hdpi.Attribute("value").Value;
+                        _logger.LogDebug("Found DpiAwareness Setting with {value}", hdpiValue);
+                    }
+                }
+            }
+
+            return hdpiValue;
+        }
+
+        private async Task UpdateHighDPISetting(IProject project)
+        {
+            var programFile = project.FindFiles("Program.cs").FirstOrDefault();
+            if (programFile is not null && File.Exists(programFile))
+            {
+                var programFileContent = File.ReadAllLines(programFile);
+
+                if (!programFileContent.Contains("Application.SetHighDpiMode"))
+                {
+                    var hdpiValue = ProcessAppConfigFile(project);
+                    try
+                    {
+                        using var outputStream = File.Create(programFile, BufferSize, FileOptions.Asynchronous | FileOptions.SequentialScan);
+
+                        await StreamHelpers.CopyStreamWithNewLineAdded(programFileContent, outputStream, "Application.EnableVisualStyles()", SR.Format("Application.SetHighDpiMode(HighDpiMode.{0});", hdpiValue)).ConfigureAwait(false);
```

Comment:
This method call is interesting. Can you move the string to constants with descriptive names so it's clear what the purpose of each is?

Reply:
Will WinForms apps always have a call to Application.EnableVisualStyles() that we can key off of?